compiler-rt/lib/asan/asan_win_new_scalar_align_nothrow_thunk.cpp

```diff
@@ -0,0 +1,34 @@
+//===-- asan_win_new_scalar_align_nothrow_thunk.cc ------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file is a part of AddressSanitizer, an address sanity checker.
+//
+// Windows-specific user-provided new/delete operator detection and fallback.
+//===----------------------------------------------------------------------===//
+#include "asan_win_new_delete_thunk_common.h"
+
+// see diagram in asan_win_new_delete_thunk_common.h for the ordering of the
+// new/delete fallbacks.
+
+extern "C" void* __cdecl __asan_new_align_nothrow(size_t size,
```

Comment:
Why you don't want to move this into header?

Reply:
Maybe you can declare all of them in header using?
#include "asan_interface.inc"

Reply:
Just because each individual TU is their only callsite, so there's no need to have all of their declarations in a header. If someone needs to call them from someplace else they can move them then.

If you feel strongly about it I can move them though. I thought you just wanted the big diagrams centralized.

Reply:
Yes, I think it would be better.
And I think there is high prob. that it would be moved by some random refactoring later.
So I don't see good reasons no to do that right now.

Other than that the patch is LGTM.